ng-dev/release/publish/test/release-notes/generation.spec.ts

```diff
@@ -38,6 +38,65 @@ describe('release notes generation', () => {
   });
 
   describe('changelog', () => {
+    describe('grouping', () => {
+      it('should group commits by scope by default', async () => {
+        SandboxGitRepo.withInitialCommit(githubConfig)
+          .createTagForHead('startTag')
+          .commit('fix(@angular-devkit/core): commit #1')
+          .commit('fix(@angular-devkit/core): commit #2')
+          .commit('fix(@angular-devkit/test): commit #3')
+          .commit('fix(@angular-devkit/test): commit #4');
+
+        const releaseNotes = await ReleaseNotes.forRange(parse('13.0.0'), 'startTag', 'HEAD');
+
+        expect(await releaseNotes.getChangelogEntry()).toMatch(changelogPattern`
+          # 13.0.0 <..>
+          ### @angular-devkit/core
+          | Commit | Description |
+          | -- | -- |
+          | <..> | fix: commit #2 |
+          | <..> | fix: commit #1 |
+          ### @angular-devkit/test
+          | Commit | Description |
+          | -- | -- |
+          | <..> | fix: commit #4 |
+          | <..> | fix: commit #3 |
+          ## Special Thanks:
+        `);
+      });
+
+      it('should support custom grouping', async () => {
+        SandboxGitRepo.withInitialCommit(githubConfig)
+          .createTagForHead('startTag')
+          .commit('fix(cdk/a11y): commit #1')
+          .commit('fix(cdk/platform): commit #2')
+          .commit('fix(material/autocomplete): commit #3')
+          .commit('fix(material/slide-toggle): commit #4');
+
+        releaseConfig.releaseNotes = {
+          // Group commits in the release notes by package name.
+          commitToGroup: (commit) => commit.scope.split('/', 1)[0],
+        };
+
+        const releaseNotes = await ReleaseNotes.forRange(parse('13.0.0'), 'startTag', 'HEAD');
+
+        expect(await releaseNotes.getChangelogEntry()).toMatch(changelogPattern`
+          # 13.0.0 <..>
+          ### cdk
+          | Commit | Description |
+          | -- | -- |
+          | <..> | fix: commit #2 |
+          | <..> | fix: commit #1 |
+          ### material
+          | Commit | Description |
+          | -- | -- |
+          | <..> | fix: commit #4 |
+          | <..> | fix: commit #3 |
```

Comment:
@josephperrott 2de52df6523fe49d593a7b27e3c0cbe042adbd0d removed the commit scope from commits. This now brings up another issue for COMP if we'd group by package. It would not be clear what scope the `fix` targets. 

Any idea on how we can make this more configurable?

Reply:
I suppose we would want to have a way to say that if a custom grouping is used, then the scope should be included in the commit description?

```md
### material
| Commit | Description |
| -- | -- |
| <..> | fix(material/dialog): commit #4 |
```

We could make the template aware of if custom grouping is being used and then include the scope if appropriate.

Have a method in `RenderContextData` like this (with a better name):

```ts
getCommitTypeAndOrScope(commit: Commit) {
  if (this.commitToGroup === undefined) {
    return commit.type;
  }
  if (commit.scope === undefined) {
    return commit.type;
  }
  return `${commit.type}(${commit.scope})`
}
```


---

Though I imagine that we actually would want to be able to control what the scope looks like then so that we could have the common part of the group removed if appropriate.

```md
### material
| Commit | Description |
| -- | -- |
| <..> | fix(dialog): commit #4 |
```

---

I am struggling to think of a way to make this type of thing possible without a second property for defining something like this.

But we could extend that method above to use this if provided in the configuration:

```ts
getCommitTypeAndOrScope(commit: Commit) {
  if (this.commitToGroup === undefined) {
    return commit.type;
  }
  if (this.commitScopeTransformer !== undefined) {
    return this.commitScopeTransformer(commit);
  }
  if (commit.scope === undefined) {
    return commit.type;
  }
  return `${commit.type}(${commit.scope})`
}
```

Reply:
Implemented as discussed offline.